src/pages/docs/chat/api/javascript/messages.mdx

```diff
@@ -0,0 +1,371 @@
+
+#### MessageMetadata | Metadata <a id="metadata"/>
+
+Metadata enables attaching extra info to a message. Uses include animations, effects, or simply to link it to other resources such as images or relative points in time.
+
+`MessageMetadata: Record<string, unknown>`
+
+#### MessageHeaders | Headers <a id="headers"/>
+
+Headers enable attaching extra info to a message. Uses include linking to a relative point in time of a livestream video or flagging this message as important or pinned.
+
+`MessageHeaders: Record<string, number | string | boolean | null | undefined>`
+
+### Returns
+
+Returns a promise. On success, the promise is fulfilled with the [`Message`](/docs/chat/api/javascript/message) object that was sent. On failure, the promise is rejected with an [`ErrorInfo`](/docs/chat/api/javascript/error-info).
+
+<Code>
+```javascript
+const message = await room.messages.send({
+  text: 'Hello, world!',
+  metadata: { priority: 'high' },
+  headers: { timestamp: Date.now() }
+});
+```
+</Code>
+
+---
+
+## Update a message <a id="update"/>
+
+`room.messages.update()`
+
+Update a message in the chat room.
+
+Note that the Promise may resolve before or after the updated message is received from the room. This means you may see the update that was just sent in a callback to `subscribe()` before the returned promise resolves.
+
+The [`Message`](/docs/chat/api/javascript/message) instance returned by this method is the state of the message as a result of the update operation. If you have a subscription to message events via `subscribe()`, you should discard the message instance returned by this method and use the event payloads from the subscription instead.
+
+This method uses PUT-like semantics: if headers and metadata are omitted from the [`updateParams`](#updateparams), then the existing headers and metadata are replaced with the empty objects.
+
+`update(serial: Serial, updateParams: UpdateMessageParams, details?: OperationDetails): Promise<Message>`
+
+### Parameters
+
+Use the following parameters:
+
+| Parameter | Description | Type |
+| --------- | ----------- | ---- |
+| `serial` | A string or object that conveys the serial of the message to update. | [`Serial`](#serial) |
+| `updateParams` | The parameters for updating the message. | [`UpdateMessageParams`](#updatemessageparams) |
+| `details` | *Optional* Optional details to record about the update action. | [`OperationDetails`](#operationdetails) |
+
+#### Serial
```

Comment:
This appears in a few places - worth putting it in its own place so we only say it once (or, dare I say, using a partial)?

Reply:
Hah... No MDX partials unless we write a component for it. It's in 2 places (`messages` and `messageReactions`) - do you think that's worth pulling into its own page? 

Alternatively we could do some sort of 'shared interfaces and types' page but I thought it might look a bit... odd.

Reply:
Probably not yet, I think we can leave it where it is for now.